src/modules/sync/index.tsx

```diff
@@ -147,6 +169,54 @@ export function ContentTabSync( { selectedSite }: { selectedSite: SiteDetails }
 		}
 	};
 
+	const handleLaunchSite = () => {
+		setModalState( ( prev ) => ( { ...prev, mode: 'push' } ) );
+		dispatch( connectedSitesActions.openModal() );
```

Comment:
I wonder if the modalState could be part of the Redux slice and the openModal could receive the `mode` as parameter.